### PR TITLE
feat: mqtt ping timer

### PR DESCRIPTION
**Pull request by YoDaMa**
related to: #417 and #537

### CODE REVIEW

File: e2etests/test/set_ping_timeout.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  describe('ConnectionString', function() {
+    it('does some stuff', function(testCallback) {
+      var connectionString = DeviceConnectionString.createWithSharedAccessKey(hostName, testDeviceId, testDeviceKey);
+      var deviceClient = DeviceClient.fromConnectionString(connectionString, Mqtt);
+      deviceClient.setOptions({'keepalive': 50})
+      deviceClient.sendEvent(new Message('testMessage'), function(err) {
+        if(err instanceof UnauthorizedError || err instanceof NotConnectedError) {
+          testCallback();
+        } else {
+          testCallback(err);
```
Comment (anthonyvercolano) [inline e2etests/test/set_ping_timeout.js]: I imagine you're going to add a bit more here?  Not sure how you plan to e2e this.

File: e2etests/test/set_ping_timeout.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  after('Destroys the test device', function(afterCallback) {
+    var registry = Registry.fromConnectionString(hubConnectionString);
+    registry.delete(testDeviceId, afterCallback);
+  });
+
+  describe('ConnectionString', function() {
+    it('does some stuff', function(testCallback) {
+      var connectionString = DeviceConnectionString.createWithSharedAccessKey(hostName, testDeviceId, testDeviceKey);
+      var deviceClient = DeviceClient.fromConnectionString(connectionString, Mqtt);
+      deviceClient.setOptions({'keepalive': 50})
```
Comment (anthonyvercolano) [inline e2etests/test/set_ping_timeout.js]: >deviceClient.setOptions({'keepalive': 50}) [](start = 6, length = 42)

Is this something that we have to make specific to mqtt?  Can it be a general keepalive that amqp could eventually implement?

File: common/transport/mqtt/devdoc/mqtt_base_requirements.md [hunk 3]
```diff
@@ -46,6 +46,8 @@ The `connect` method establishes a connection with the server using the config o
 
 **SRS_NODE_COMMON_MQTT_BASE_18_002: [** The `connect` method shall set the `wsOptions.agent` option based on the `mqtt.webSocketAgent` object passed in the `options` structure via the `setOptions` function. **]**
 
+**SRS_NODE_COMMON_MQTT_BASE_18_002: [** The `connect` method shall set the `wsOptions.agent` option based on the `mqtt.webSocketAgent` object passed in the `options` structure via the `setOptions` function. **]**
+
```
Comment (anthonyvercolano) [inline common/transport/mqtt/devdoc/mqtt_base_requirements.md]: This doesn't seem to have anything to do with ping timers.

Comment (YoDaMa) [reply common/transport/mqtt/devdoc/mqtt_base_requirements.md]: good catch. shouldn't look like that.

**Pull request merged**